File: aspnet-with-ef/HuHao.BlogService.Domain/Core/Exceptions/EntityExistedException.cs
using System;

namespace HuHao.BlogService.Domain.Core.Exceptions
{
    public class EntityExistedException : DomainException
    {
        public EntityExistedException(string message) : base(message)
        {
        }

        public EntityExistedException(Type type, IEntityId id) :
            this($"the {type.Name.ToLower()} with id {id} was existed")
        {
        }
    }
}

File: next-version/blog-service/src/main/java/dev/huhao/samples/ddd/blogservice/domain/common/EntityExistedException.java
package dev.huhao.samples.ddd.blogservice.domain.common;


import dev.huhao.samples.ddd.blogservice.domain.concepts.Entity;

import java.util.UUID;

public class EntityExistedException extends DomainException {
    public EntityExistedException(String message) {
        super(message);
    }

    public <T extends Entity> EntityExistedException(Class<T> entityClass, UUID id) {
        super("the " + entityClass.getSimpleName().toLowerCase() + " with id " + id + " was existed");
    }
}
